src/style/style_layer/custom_style_layer.js

```diff
@@ -4,8 +4,9 @@ import StyleLayer from '../style_layer.js';
 import type Map from '../../ui/map.js';
 import assert from 'assert';
 import type {ValidationErrors} from '../validate_style.js';
+import type {ProjectionSpecification} from '../style-spec/types.js';
 
-type CustomRenderMethod = (gl: WebGLRenderingContext, matrix: Array<number>) => void;
+type CustomRenderMethod = (gl: WebGLRenderingContext, matrix: Array<number>, projection?: ProjectionSpecification, projectionToMercatorMatrix?: Array<number>, projectionToMercatorTransition?: number) => void;
```

Comment:
Following a bit the conversation from https://github.com/mapbox/mapbox-gl-js/pull/12182#discussion_r1072935701, and since we're adding a few new parameters in that interface, it might make sense to wrap it under a single variable `projectionParams` so changes don't break client-side code if we need to remove, add or re-order new parameters to that interface. It would become:

```js
type CustomRenderMethod = (gl: WebGLRenderingContext, matrix: Array<number>, projectionParams: ProjectionParams) => void;
```

```js
export type ProjectionParams = {
  projection?: ProjectionSpecification,
  projectionToMercatorMatrix?: Array<number>, 
  projectionToMercatorTransition?: number
};
```

Reply:
this can be implemented in a follow-up PR along with the fix for https://github.com/mapbox/mapbox-gl-js/pull/12182#issuecomment-1401668623